pkg/controllers/consolidation/controller.go

```diff
@@ -88,25 +89,11 @@ func NewController(clk clock.Clock, kubeClient client.Client, provisioner *provi
 	}
 }
 
-func (c *Controller) Register(ctx context.Context, m manager.Manager) error {
-	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).Named("consolidation"))
-	go func() {
-		select {
-		case <-ctx.Done():
-			return
-		case <-m.Elected():
-			for {
-				select {
-				case <-ctx.Done():
-					logging.FromContext(ctx).Infof("Shutting down")
-					return
-				case <-c.clock.After(pollingPeriod):
-					_, _ = c.Reconcile(ctx, reconcile.Request{})
-				}
-			}
-		}
-	}()
-	return nil
+func (c *Controller) Register(_ context.Context, m manager.Manager) error {
+	return controller.NewSingletonManagedBy(m).
+		Named("consolidation").
+		LeaderElected().
```

Comment:
Any reason we can't block all of these on leader election and simply not expose it?